### PR TITLE
DRILL-6862: Update Calcite to 1.18.0

### PR DESCRIPTION
**Pull request by ihuzenko**
1. Moved Calcite dependency from profile hadoop-default to general dependency managment
2. Updated Calcite version to 1.18.0-drill-r0 and Avatica version to 1.13.0
3. Hook.REL_BUILDER_SIMPLIFY moved to static block, cause now it can't be removed (fixes DRILL-6830)
4. Removed WrappedAccessor, since it was workaround fixed in CALCITE-1408
5. Fixed setting of multiple options in TestBuilder
6. Timstampadd type inference aligned with CALCITE-2699
7. Dependency update caused 417 kB increase of jdb-all jar size, so the maxsize limit was
   increased from 39.5 to 40 MB
8. Added test into TestDrillParquetReader to ensure that DRILL-6856 was
   fixed by Calcite update

### CODE REVIEW

**Comment by ihuzenko** [comment]
@vvysotskyi  could you review please ?

File: exec/java-exec/src/main/codegen/data/Parser.tdd [hunk 1]
```diff
@@ -89,8 +89,766 @@
   ]
 
   # List of keywords from "keywords" section that are not reserved.
-  nonReservedKeywords: [
-  ]
+  # Added due to comment under CALCITE-2405.
```
Comment (vvysotskyi) [inline exec/java-exec/src/main/codegen/data/Parser.tdd]: Could you please rephrase this comment:
```
Copied from calcite-core config.fmpp
For details please see comment under CALCITE-2405.
```

File: exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet2/TestDrillParquetReader.java [hunk 2]
```diff
@@ -434,4 +434,20 @@ public void testLogicalIntTypes5() throws Exception {
...
+    Arrays.fill(expectedRowValues, 0, expectedRowValues.length - 2, true);
+    testBuilder()
+        .sqlQuery("SELECT bool_col " +
+            "FROM cp.`parquet2/0_0_0.parquet` " +
+            "WHERE bool_col IS true OR bool_col IS null " +
+            "ORDER BY bool_col")
+        .ordered()
+        .baselineColumns("bool_col")
+        .baselineValuesForSingleColumn(expectedRowValues)
+        .build().run();
```
Comment (vvysotskyi) [inline exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet2/TestDrillParquetReader.java]: `.build().run()` -> `.go()`

File: exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet2/TestDrillParquetReader.java [hunk 3]
```diff
@@ -434,4 +434,20 @@ public void testLogicalIntTypes5() throws Exception {
...
   }
 
+  @Test // DRILL-6856
+  public void testIsTrueOrNullCondition() throws Exception {
+    Object[] expectedRowValues = new Object[9];
+    // 7 true values and 2 null values in the end
+    Arrays.fill(expectedRowValues, 0, expectedRowValues.length - 2, true);
+    testBuilder()
+        .sqlQuery("SELECT bool_col " +
+            "FROM cp.`parquet2/0_0_0.parquet` " +
```
Comment (vvysotskyi) [inline exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet2/TestDrillParquetReader.java]: Is it possible to use an existing table or at least create it in the test?

**Comment by ihuzenko** [comment]
@vvysotskyi  I've addressed your comments, please check again.

**Pull request closed without merging**